src/components/views/login/RegistrationForm.js

```diff
@@ -50,31 +57,14 @@ module.exports = React.createClass({
             email: this.props.defaultEmail,
             username: this.props.defaultUsername,
             password: null,
-            passwordConfirm: null
+            passwordConfirm: null,
+            fieldValid: {}
         };
     },
 
     onSubmit: function(ev) {
         ev.preventDefault();
 
-        var pwd1 = this.refs.password.value.trim();
-        var pwd2 = this.refs.passwordConfirm.value.trim()
-
-        var errCode;
-        if (!pwd1 || !pwd2) {
-            errCode = "RegistrationForm.ERR_PASSWORD_MISSING";
-        }
-        else if (pwd1 !== pwd2) {
-            errCode = "RegistrationForm.ERR_PASSWORD_MISMATCH";
-        }
-        else if (pwd1.length < this.props.minPasswordLength) {
-            errCode = "RegistrationForm.ERR_PASSWORD_LENGTH";
-        }
-        if (errCode) {
-            this.props.onError(errCode);
-            return;
-        }
-
```

Comment:
We're removing the checks from "submit" time and moving them to be recalculated on the fly which is cool, but we then need to check in `onSubmit` that there are no invalid fields. I don't see this check being made, so what's stopping me from just hitting Register with invalid fields?